test/functional/feature_rbf.py

```diff
@@ -624,6 +627,30 @@ def test_no_inherited_signaling(self):
         assert_equal(True, self.nodes[0].getmempoolentry(optin_parent_tx['txid'])['bip125-replaceable'])
         assert_raises_rpc_error(-26, 'txn-mempool-conflict', self.nodes[0].sendrawtransaction, replacement_child_tx["hex"], 0)
 
+    def test_replacement_relay_fee(self):
+        wallet = MiniWallet(self.nodes[0])
+        wallet.scan_blocks(start=77, num=1)
+        utxo = wallet.get_utxo()
+
+        initial_transaction = CTransaction()
+        initial_transaction.vin = [CTxIn(COutPoint(int(utxo['txid'], 16), utxo['vout']), nSequence=0)]
+        initial_transaction.vout = [CTxOut(1 * COIN, DUMMY_P2WPKH_SCRIPT)]
+        initial_transaction.wit.vtxinwit = [CTxInWitness()]
+        initial_transaction.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
+        initial_transaction_hex = txToHex(initial_transaction)
+        self.nodes[0].sendrawtransaction(initial_transaction_hex, 0)
+
+        # Equal fee, but the replacement does not provide a relay
+        # fee conforming to node's `incrementalrelayfee` policy of 1000 sat per KB.
+        replacement_transaction = CTransaction()
+        replacement_transaction.vin = [CTxIn(COutPoint(int(utxo['txid'], 16), utxo['vout']), nSequence=1)]
+        replacement_transaction.vout = [CTxOut(1 * COIN, DUMMY_P2WPKH_SCRIPT)]
+        replacement_transaction.wit.vtxinwit = [CTxInWitness()]
+        replacement_transaction.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
+        replacement_transaction_hex = txToHex(replacement_transaction)
```

Comment:
So now this is the very same test as in `test_simple_doublespend`:
https://github.com/bitcoin/bitcoin/blob/7317e14a44c6efc545e6fb9bcedee7174e93a8fa/test/functional/feature_rbf.py#L133-L148

I think (untested) it would be caught in:
https://github.com/bitcoin/bitcoin/blob/7317e14a44c6efc545e6fb9bcedee7174e93a8fa/src/validation.cpp#L830-L838
Which is already a covered branch.

You need to keep the lower feerate to get to the yet-uncovered branch of:
https://github.com/bitcoin/bitcoin/blob/7317e14a44c6efc545e6fb9bcedee7174e93a8fa/src/validation.cpp#L892-L900

And if you modified that because of my previous brainfart i'm sorry :no_mouth:

Reply:
> You need to keep the lower feerate to get to the yet-uncovered branch of:

Oh yes effectively, do you mean the higher feerate ? 

At 58cdabe6, feerates are the following:

```
 node0 2021-06-24T15:51:23.735515Z [httpworker.0] [validation.cpp:831] [PreChecks] newFeeRate 583.33333345 BTC/kvB 
 node0 2021-06-24T15:51:23.735525Z [httpworker.0] [validation.cpp:832] [PreChecks] oldFeeRate 583.33333333 BTC/kvB 
```

I think it's a +1 to have different `reject_reason` :p